src/Microsoft.ML.Transforms/Text/WrappedTextTransformers.cs

```diff
@@ -119,4 +122,747 @@ private static TransformWrapper MakeTransformer(IHostEnvironment env, (string in
+        /// takes tokenized text as input while <see cref="WordHashBagEstimator"/> tokenizes text internally.
+        /// </summary>
+        /// <param name="env">The environment.</param>
+        /// <param name="columns">Pairs of columns to compute bag of word vector.</param>
+        /// <param name="hashBits">Number of bits to hash into. Must be between 1 and 30, inclusive.</param>
+        /// <param name="ngramLength">Ngram length.</param>
+        /// <param name="skipLength">Maximum number of tokens to skip when constructing an ngram.</param>
+        /// <param name="allLengths">Whether to include all ngram lengths up to <paramref name="ngramLength"/> or only <paramref name="ngramLength"/>.</param>
+        /// <param name="seed">Hashing seed.</param>
+        /// <param name="ordered">Whether the position of each source column should be included in the hash (when there are multiple source columns).</param>
+        /// <param name="invertHash">Limit the number of keys used to generate the slot name to this many. 0 means no invert hashing, -1 means no limit.</param>
+        public NgramHashEstimator(IHostEnvironment env,
+            (string[] inputs, string output)[] columns,
+            int hashBits = 16,
+            int ngramLength = 2,
+            int skipLength = 0,
+            bool allLengths = true,
+            uint seed = 314489979,
+            bool ordered = true,
+            int invertHash = 0)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(WordBagEstimator)))
+        {
+            foreach (var (input, output) in columns)
+            {
+                Host.CheckUserArg(Utils.Size(input) > 0, nameof(input));
+                Host.CheckValue(output, nameof(input));
+            }
+
+            _columns = columns;
+            _hashBits = hashBits;
+            _ngramLength = ngramLength;
+            _skipLength = skipLength;
+            _allLengths = allLengths;
+            _seed = seed;
+            _ordered = ordered;
+            _invertHash = invertHash;
+        }
+
+        public override TransformWrapper Fit(IDataView input)
+        {
+            // Create arguments.
+            var args = new NgramHashTransform.Arguments
+            {
+                Column = _columns.Select(x => new NgramHashTransform.Column { Source = x.inputs, Name = x.output }).ToArray(),
+                HashBits = _hashBits,
+                NgramLength = _ngramLength,
+                SkipLength = _skipLength,
+                AllLengths = _allLengths,
+                Seed = _seed,
+                Ordered = _ordered,
+                InvertHash = _invertHash
+            };
+
+            return new TransformWrapper(Host, new NgramHashTransform(Host, args, input));
+        }
+    }
+
+    /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+    public sealed class LdaEstimator : TrainedWrapperEstimatorBase
```

Comment:
>LdaEstimator [](start = 24, length = 12)

Please only review LDA related classes. The other text transform changes are being reviewed in #953. I will merge the branches once approved.

Reply:
merged.

---
In reply to: [219356958](https://github.com/dotnet/machinelearning/pull/972#discussion_r219356958) [](ancestors = 219356958)